### ISSUE TITLE
Unicode Error for Hindi transcription 

### ISSUE THREAD

**Issue by rahulshivajipawar** [ggerganov/whisper.cpp#1700]
When doing transcription in Hindi for a file, I encounter invalid unicode character. 

<img width="753" alt="Screenshot 2023-12-29 at 8 29 09 PM" src="https://github.com/ggerganov/whisper.cpp/assets/7852108/340f9bab-4299-4103-9055-fa5a9db4e989">

I have noticed this with many Hindi files. 

Used whisper-large-v2 mode for inference on CPU. Have noticed the same issue when inferencing on GPU as well. 

I am guessing the issue is: whisper model token output (BPE encoded) is not getting correctly mapped to unicode characters.

**Comment by rahulshivajipawar** [comment]
This is an example of a file on which it fails - https://drive.google.com/file/d/1_BFuNOAqM3yv4P2A0i8KOT_RZ6LnYSCt/view?usp=sharing

**Comment by bobqianic** [comment]
I'll explore this further. There might be an issue with the tokenizer. 
Take a look at this discussion for more insight: https://github.com/ggerganov/whisper.cpp/pull/1313#discussion_r1381312680

**Comment by rahulshivajipawar** [comment]
> I'll explore this further. There might be an issue with the tokenizer. Take a look at this discussion for more insight: [#1313 (comment)](https://github.com/ggerganov/whisper.cpp/pull/1313#discussion_r1381312680)

Looked at the issue you referenced, looks like it is similar to this https://github.com/ggerganov/whisper.cpp/pull/1313#issuecomment-1799347484

**Comment by rahulshivajipawar** [comment]
Another side observation - I have found the accuracy for Hindi transcription via whisper.cpp to be much lower than when using openai API directly.

**Comment by bobqianic** [comment]
> Another side observation - I have found the accuracy for Hindi transcription via whisper.cpp to be much lower than when using openai API directly.

Agreed. You don't even need to use their API. Just by using `pip install openai-whisper` and running large-v2 locally, you can achieve better results than whisper.cpp in most languages apart from English.
